futures-util/src/async_await/select.rs

```diff
@@ -47,16 +47,18 @@ macro_rules! select {
             )*
         }
 
-        let __priv_res = loop {
+        let __priv_res = await!($crate::future::poll_fn(|cx| {
             $(
-                let poll_res = $crate::poll!($crate::core_reexport::mem::PinMut::new(
-                        &mut $name));
-                if let $crate::core_reexport::task::Poll::Ready(x) = poll_res {
-                    break __PrivResult::$name(x);
+                match $crate::future::Future::poll(
```

Comment:
`$crate::core_reexport::future::Future` would also work. It's IMO no different than `Poll`

Reply:
Done.